src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.State.cs

```diff
@@ -52,7 +52,7 @@ private class State
                 /// needed as the compilation an assembly came from can be GC'ed and further requests to get that
                 /// compilation (or any of it's assemblies) may produce new assembly symbols.
                 /// </summary>
-                public readonly WeakSet<ISymbol>? UnrootedSymbolSet;
+                public readonly UnrootedSymbolSet? UnrootedSymbolSet;
```

Comment:
changed to a new stronger type to separate out two separate classes of symbols.  That separation will be explained below.